### ISSUE TITLE
Critical alerts, or otherwise a different type of alerts

### ISSUE THREAD

**Issue by norayna** [JohanDegraeve/xdripswift#429]
Hello, I have an enhancement request - apologies if anything in my request is incorrect, I don't really understand Swift; also I had a look through existing issues and I don't think there was a similar one.

Here's the situation:
- I use the app with an Apple Watch 
- all the app alerts (high glucose, low glucose, missed readings etc) do come up as a notification on my Apple Watch
- however, there is never a watch vibration for those notifications
- as I understand it, this is probably because of the type of iOS notification being used
- I'm guessing the type of notification used in Shuggah is some sort of media notification, because that way it can have a custom sound depending on the user's choice in settings
- However, it seems that this type of notification does not ellicit a watch vibration by default

- **It would be a much better user experience if glucose alerts would vibrate my watch. That way I would notice my high/low blood sugar even if my phone was on silent, without disrupting my environment.**
- As an extra, it would be nice to have 2 vibration options: a single vibration like on a normal notification from any other app, or a continuous vibration that waits for snoozing to stop.

**Comment by paulplant** [comment]
What you are describing is the normal way for WatchOS to handle mirrored iOS notifications. 

This isn't actually an issue or bug with xDrip4iOS per se.

The Watch app will be re-written when time allows to incorporate native notifications which will give a much richer experience for alarms etc (sorry, but I don't know if this will be weeks or months).

Are you ok if we close this issue now (as it's just a feature request rather than a bug/issue)?

**Comment by norayna** [comment]
Ah yes I see what you mean about the default WatchOS mirroring behaviour, however I feel like the issue can be resolved without making changes to the Watch app because pretty much all other apps cause a vibration notification on the watch. This is where my lack of knowledge of iOS and WatchOS fails me, but I believe the type of notification used by Shuggah must be some type of special case that doesn't cause a vibration on WatchOS.

So I think there is a possibility that it's an easy change, but there's also a possibility that it requires a big refactor. And unfortunately I've not been able to figure out exactly what the change would involve.

Yes I'm happy for the issue to be closed if you think it's more of a feature request, would you like me to describe and submit this somewhere else as a feature request? I totally understand if you'd rather use issues just for bugs.

**Comment by crsohr** [comment]
Notifications vibrate on the watch when the sound is enabled.  Have you changed your settings to mute the alert?

**Comment by OliNix** [comment]
To reinvigorate this, I have worked out the logic:

**IF("override mute" = ON) AND Apple Watch connected (and being worn)**
- Phone: Audio alerts will always sound. Visual Notification will be displayed or stored in Notification Centre
- Watch: No audio alert. No tactile (vibration) alert. Visual Notification will be displayed if wrist raised or stored in Notification Centre if not raised.

**IF("override mute" = OFF) AND Apple Watch connected (and being worn)**
- Phone: Audio alerts will never sound when phone is locked (regardless if phone is on silent or not). Audio alert sounds will follow silent setting of phone if alert displays while phone is being used. Visual Notification will be displayed or stored in Notification Centre
- Watch: Ding audio alert. Prominent tactile (vibration) alert. Visual Notification will be displayed if wrist raised or stored in Notification Centre if not raised.

**IF("override mute" = OFF) AND Apple Watch NOT connected**
- Phone: Audio alert sounds will follow silent setting of phone. Visual Notification will be displayed or stored in Notification Centre
- Watch: N/A

**Summary**
In summary, the alerts are not setup correctly for IF("override mute" = OFF) AND Apple Watch connected (and being worn), they "should" follow the same logic as phone calls or text message alerts, where both the phone alerts as well as the watch, silent settings are managed per device (i.e you can mute the phone or watch independently of one another):

- Phone: Audio alert sounds will follow silent setting of phone. Visual Notification will be displayed or stored in Notification Centre
- Watch: Will mirror phone alerts.  Audio is dependent on watch mute setting. Tactile and visual alerts always display unless in theatre mode.
